rfcs/text/refactoring-date-component.md

```diff
@@ -0,0 +1,330 @@
+
+# Alternatives
+
+One alternative to carrying out the proposed in this document is that we do not do it at all. We continue to maintain the existing component as is, fix the bugs and look to introduce the feature requests raised. This is an option in the short-term, but the code is already bloated significantly and there are a number of regressions some of which have not been picked up by our cypress or chromatic tests. Therefore, it is likely that we will be revisiting the need to rewrite the component sooner rather than later.
+
+Another option is that we refactor the existing component to simplify the code and fix the issues already raised. However, we would still need to introduce breaking changes as we plan to refactor all class-based input components to be functional anyway. As well as this, refactoring could likely prove more complex than rewriting the component: some of the legacy code outdates any of the current Carbon team and is tightly coupled with older projects.
+
+In terms of alternate designs, we could look to alter the value emitted on event handling, other UI libraries emit a Date object for example. This was rejected as it would mean we would make migrating from the old implementation to the new component more difficult and may mean move away from what was defined in previous RFC documents. Currently our consuming projects expect to pass a string value and to receive one back, if we keep this pattern then we are not putting additional load on to them with respect to formatting the value from Date to string.
+
+There are a number of date management solutions available as an alternative to `date-fns`: [Luxon](https://moment.github.io/luxon/); [Dayjs](https://day.js.org/); [js-Joda](https://js-joda.github.io/js-joda/); or using the Date object directly instead or a library. An additional alternative would be to integrate an abstraction for any of these potential solutions, for example [date-io](https://github.com/dmtrKovalenko/date-io) would support this. Of these options, `date-fns` is the most lightweight and will require the least integration into the new component. It is either the chosen or default option for many other UI libraries. For our requirements, it is unlikely we will need to surface an abstraction at this point as we will not be requiring implementation teams to do any configuration past what is defined in the interface and what they will need to pass in via the locale. If the need to introduce an abstraction arose it would not be too complex to refactor it to support this without introducing a breaking change.
```

Comment:
```suggestion
There are a number of date management solutions available as an alternative to `date-fns`: [Luxon](https://moment.github.io/luxon/); [Dayjs](https://day.js.org/); [js-Joda](https://js-joda.github.io/js-joda/); or using the Date object directly instead of a library. An additional alternative would be to integrate an abstraction for any of these potential solutions, for example [date-io](https://github.com/dmtrKovalenko/date-io) would support this. Of these options, `date-fns` is the most lightweight and will require the least integration into the new component. It is either the chosen or default option for many other UI libraries. For our requirements, it is unlikely we will need to surface an abstraction at this point as we will not be requiring implementation teams to do any configuration past what is defined in the interface and what they will need to pass in via the locale. If the need to introduce an abstraction arose it would not be too complex to refactor it to support this without introducing a breaking change.
```